cmd/yurt-device-controller/app/core.go

```diff
@@ -0,0 +1,197 @@
+/*
+Copyright 2022 The OpenYurt Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes"
+	clientgoscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/klog/v2"
+	"k8s.io/klog/v2/klogr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/healthz"
+
+	devicev1alpha1 "github.com/openyurtio/device-controller/api/v1alpha1"
+	"github.com/openyurtio/device-controller/cmd/yurt-device-controller/options"
+	"github.com/openyurtio/device-controller/controllers"
+	controllerutil "github.com/openyurtio/device-controller/controllers/util"
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+func init() {
+	utilruntime.Must(clientgoscheme.AddToScheme(scheme))
+	utilruntime.Must(devicev1alpha1.AddToScheme(scheme))
+	//+kubebuilder:scaffold:scheme
+}
+
+func NewCmdYurtDeviceController(stopCh <-chan struct{}) *cobra.Command {
+	yurtDeviceControllerOptions := options.NewYurtDeviceControllerOptions()
+	cmd := &cobra.Command{
+		Use:   "yurt-device-controller",
+		Short: "Launch yurt-device-controller",
+		Long:  "Launch yurt-device-controller",
+		Run: func(cmd *cobra.Command, args []string) {
+			cmd.Flags().VisitAll(func(flag *pflag.Flag) {
+				klog.V(1).Infof("FLAG: --%s=%q", flag.Name, flag.Value)
+			})
+			if err := options.ValidateOptions(yurtDeviceControllerOptions); err != nil {
+				klog.Fatalf("validate options: %v", err)
+			}
+			Run(yurtDeviceControllerOptions, stopCh)
+		},
+	}
+
+	yurtDeviceControllerOptions.AddFlags(cmd.Flags())
+	return cmd
+}
+
+func Run(opts *options.YurtDeviceControllerOptions, stopCh <-chan struct{}) {
+	ctrl.SetLogger(klogr.New())
+	cfg := ctrl.GetConfigOrDie()
+
+	mgr, err := ctrl.NewManager(cfg, ctrl.Options{
+		Scheme:                 scheme,
+		MetricsBindAddress:     opts.MetricsAddr,
+		Port:                   9443,
```

Comment:
yurt-device-controller has't involve any webhook,  seems no need to set this Port field.

Reply:
The port field has been removed.